### DISCUSSION TITLE
Record Internet Radio

### DISCUSSION THREAD

**Discussion by xstable** [zaps166/QMPlay2#738] [Q&A]
I came to QMPlay2 as I was searching for a solution to record audio-streams like I did decades before with winamp. 

I've found that QMPlay2 has both... Internet-radio and the ability to record "download" them. 
But I wonder how does it work to have the files splittet by song and titled by the recognized Radio-Text.

If I play an Internet-Radio-Station and start to download it... it never stop the recording. And if I stop the recording it's marked as "aborted download". 

So can anybody give me a helping hand... if - and how - I can download songs from internet-radio which are saved one by one with e nice filename on my disk?

**Comment by zaps166** [comment]
> "aborted download"

You can still access the file by double-click the aborted item. Also the file exists on your disk (probably need to add file suffix depends on stream type on Windows to easily open it).

---

You can also record by pressing Shift+C (or click Player->Recording).

---

> which are saved one by one with e nice filename on my disk?

Not yet possible, however downloading OGG radio streams will split titles inside a file and QMPlay2 will load them as playlist.

**Reply by xstable** [reply]
Yes, i can play this aborted record, and it's a regulare downloaded file at my disk.
I only thought it would cut and name the files by radio-text... but good to know if it's possible with OGG.

Could you give me a hint how to get this to work?
Maybe an example Ogg-Stream to test it?

**Reply by zaps166** [reply]
Sorry I didn't notice your message.

>  Maybe an example Ogg-Stream to test it?

You can sort the radio results by "Stream info". You'll find some OGG streams there.

**Reply by xstable** [reply]
I still don't see how it should work. 
Can you give a short explanation with some screenshots please? THx.

**Reply by zaps166** [reply]
You can record on 2 ways:
- Record the stream - you'll write to disk the stream which is currently playing with file suffix included (to start/stop use Shift+C hotkey or Player->Recording), the example filename will be `QMPlay2_rec_00001.aac`.
- Download the stream (Right-click->Extensions->Download) - it starts downloading a stream to a file, it's independent to playback, you can see it in "Downloader" tab, in some cases you have to put file suffix (file extension) and name may not nice - just rename the file as you wish, QMPlay2 will still play this without file suffix, double-click in "Downloader" tab to quickly play it in QMPlay2.

---

> if - and how - I can download songs from internet-radio which are saved one by one with e nice filename on my disk?

In QMPlay2, you can't.

In OGG streams (it's rare case for radio stations) most likely it'll store song titles in the file, so you'll be able to see it while playing recorded file.

Stream is recorded as-is without transcoding, so you can losslessly use FFmpeg's streamcopy to split it to file, but it's manual task to trim every song...
